### PR TITLE
Submits photo posts [pr]

### PR DESCRIPTION
**Pull request by aaronschachter**
#### What's this PR do?

Continues #443, submitting a photo post to the Rogue API via new `createPhotoPost` user heper, and starts on unit tests for photo post middleware.

#### How should this be reviewed?

Upon setting `DS_GAMBIT_CONVERSATIONS_DRAFT_SUBMISSION_ENABLED` to `true`, verify a photo post is created upon completing a photo post conversation (e.g. text `lock` to submit a photo for Friends On Lock)

#### Any background context you want to provide?

Once this goes live and the config variable is set on production, we'll be in great shape to exclude Campaign Run Id's to Rogue requests in a single app/deploy, as the Gambit Campaigns app will no longer post to Rogue.

#### Relevant tickets

https://www.pivotaltracker.com/story/show/161494958

#### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Tests added/updated for new features/bug fixes.
- [x] Tested on staging.

### CODE REVIEW

File: lib/middleware/messages/member/topics/posts/photo/create.js [hunk 1]
```diff
@@ -11,13 +11,23 @@ module.exports = function createPhotoPost() {
...
       }
 
-      // TODO: Fetch file from url property to submit a photo post.
-      const createPostRes = await helpers.user.createTextPost(req.user, {
+      if (!req.draftSubmission) {
+        throw new Error('req.draftSubmission undefined');
+      }
+
+      const values = req.draftSubmission.values;
+      const file = await helpers.util.fetchImageFileFromUrl(values.url);
```
Comment (DFurnes) [inline lib/middleware/messages/member/topics/posts/photo/create.js]: I assume this URL something that Twilio provides us in their message payload? (Just making sure I follow!)

File: test/helpers/factories/draftSubmission.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  };
+}
+
+/**
+ * @return {DraftSubmission}
+ */
+function getValidCompletePhotoPostDraftSubmission() {
+  return DraftSubmission(getRawDraftSubmissionData({
+    quantity: 4,
+    url: 'http://placekitten.com/g/300/300',
```
Comment (DFurnes) [inline test/helpers/factories/draftSubmission.js]: 🐱

File: lib/middleware/messages/member/topics/posts/photo/create.js [hunk 3]
```diff
@@ -11,13 +11,23 @@ module.exports = function createPhotoPost() {
...
+
+      const createPostRes = await helpers.user.createPhotoPost(req.user, {
         campaignId: req.topic.campaign.id,
         campaignRunId: req.topic.campaign.currentCampaignRun.id,
-        text: JSON.stringify(req.draftSubmission.toObject().values),
+        file,
+        quantity: values.quantity,
         source: req.platform,
+        text: values.whyParticipated,
+        whyParticipated: values.whyParticipated,
```
[the remark this answers is not in the document]
Comment (aaronschachter) [reply lib/middleware/messages/member/topics/posts/photo/create.js]: The `text` value is required in the Rogue API. going to be updating this middleware in a future PR to collect a different value to send here.

File: lib/middleware/messages/member/topics/posts/photo/create.js (hunk 1 above)
Comment (aaronschachter) [reply lib/middleware/messages/member/topics/posts/photo/create.js]: yup!

**Pull request merged**